astropy/io/ascii/ecsv.py

```diff
@@ -296,3 +429,19 @@ def update_table_data(self, table):
         with serialize_context_as('ecsv'):
             out = serialize.represent_mixins_as_columns(table)
         return out
+
+    def _check_multidim_table(self, table, max_ndim=1):
```

Comment:
So this would be `_max_ndim = np.inf` (or something more reasonable?)

Reply:
I used `max_ndim = None` for no limit.